WorkflowFunctionBundle/Tests/Functional/Repository/WorkflowFunctionRepositoryTest.php

```diff
@@ -0,0 +1,126 @@
+    }
+
+    /**
+     * @param array  $descriptionEntity
+     * @param array  $columns
+     * @param string $search
+     * @param array  $order
+     * @param int    $skip
+     * @param int    $limit
+     * @param int    $count
+     *
+     * @dataProvider providePaginateAndSearch
+     */
+    public function testFindForPaginateAndSearch($descriptionEntity, $columns, $search, $order, $skip, $limit, $count)
+    {
+        $worflowFunctions = $this->repository->findForPaginateAndSearch($descriptionEntity, $columns, $search, $order, $skip, $limit);
+        $this->assertCount($count, $worflowFunctions);
+    }
+
+    /**
+     * @return array
+     */
+    public function providePaginateAndSearch()
+    {
+        $descriptionEntity = $this->getDescriptionColumnEntity();
+
+        return array(
+            array($descriptionEntity, $this->generateColumnsProvider(), null, null, 0 ,5 , 2),
+            array($descriptionEntity, $this->generateColumnsProvider('validator'), null, null, 0 ,5 , 1),
+            array($descriptionEntity, $this->generateColumnsProvider('contributor'), null, null, 0 ,5 , 1),
+            array($descriptionEntity, $this->generateColumnsProvider('fakeName'), null, null, 0 ,5 , 0),
+            array($descriptionEntity, $this->generateColumnsProvider(), 'validator', null, 0 ,5 , 1),
+        );
+    }
+
+    /**
+     * test count all workflow function
+     */
+    public function testCount()
+    {
+        $worflowFunctions = $this->repository->count();
+        $this->assertEquals(2, $worflowFunctions);
+    }
+
+    /**
+     * @param array  $columns
+     * @param array  $descriptionEntity
+     * @param string $search
+     * @param int    $count
+     *
+     * @dataProvider provideColumnsAndSearchAndCount
+     */
+    public function testCountFilterSearch($descriptionEntity, $columns, $search, $count)
+    {
+        $worflowFunctions = $this->repository->countFilterSearch($descriptionEntity, $columns, $search);
+        $this->assertEquals($count, $worflowFunctions);
+    }
+
+    /**
+     * @return array
+     */
+    public function provideColumnsAndSearchAndCount(){
+        $descriptionEntity = $this->getDescriptionColumnEntity();
+
+        return array(
+            array($descriptionEntity, $this->generateColumnsProvider(), null, 2),
+            array($descriptionEntity, $this->generateColumnsProvider('validator'), null, 1),
+            array($descriptionEntity, $this->generateColumnsProvider('contributor'), null, 1),
+            array($descriptionEntity, $this->generateColumnsProvider('fakeName'), null, 0),
+            array($descriptionEntity, $this->generateColumnsProvider(), 'validator', 1),
+        );
+    }
+
+    /**
+     * Generate columns of content with search value
+     *
+     * @param string $searchName
+     *
+     * @return array
+     */
+    protected function generateColumnsProvider($searchName = '')
```

Comment:
this method is used by several method, currently is just under the last provider wich use it.